R/zchunk_LA154.Transport_china.R

```diff
@@ -0,0 +1,176 @@
+        mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>%
+        # Filtering for just NBS-fuel/sector pairs
+        filter(fuel_sector %in% paste(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBProcess, L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBMaterial)) %>%
+        select(province, EBProcess, EBMaterial, sector, fuel, year, value) %>%
+        group_by(province, year) %>%
+        # First zero out NAs in years where some values are NA but not all
+        mutate(value = replace(value, is.na(value) & sum(value, na.rm = T) != 0, 0)) %>%
+        ungroup %>%
+        # use approx_fun rule = 2 to fill out data in years where the entire province is NA
+        group_by(province, EBProcess, EBMaterial) %>%
+        mutate(value = approx_fun(year, value, rule = 2)) %>%
+        ungroup() ->
+        L154.NBS_trn_Mtce_province
+
+      # Now the province shares can be calculated
+      L154.NBS_trn_Mtce_province %>%
+        group_by(EBProcess, EBMaterial, year) %>%
+        mutate(value_share = value / sum(value, na.rm = T)) %>%
+        complete(nesting(year, sector, fuel, EBProcess, EBMaterial), province = gcamchina.PROVINCES) %>%
+        ungroup() %>%
+        # NAs were introduced where national values were 0. Replace NAs with zeros.
+        replace_na(list(value_share = 0)) %>%
+        select(province, EBProcess, EBMaterial, year, value_share) %>%
+        left_join(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh, by = c("EBProcess", "EBMaterial", "year")) %>%
+        filter(year %in% HISTORICAL_YEARS) %>%
+        mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>%
+        replace_na(list(value_share = 0)) %>%
+        mutate(value = value * value_share) %>% # Allocating across the provinces
+        select(province, UCD_sector, mode, size.class, UCD_technology, UCD_fuel, fuel, year, value) ->
+        L154.in_EJ_province_trn_m_sz_tech_F
+
+      # As a final step, aggregate by fuel and name the sector
+      # This creates the second of three output tables
+      L154.in_EJ_province_trn_m_sz_tech_F %>%
+        group_by(province, fuel, year) %>%
+        summarise(value = sum(value)) %>%
+        ungroup() %>%
+        # Adding a column named "sector" with "transportation" as the entries
+        mutate(sector = "transportation") %>%
+        select(province, sector, fuel, year, value) ->
+        L154.in_EJ_province_trn_F
+
+      # Apportion non-motorized energy consumption to provinces on the basis of population
+      # First we will create the province shares based on population
+      L100.Pop_thous_province %>%
+        complete(province, year = c(1971:2100)) %>%
+        group_by(province) %>%
+        mutate(pop = approx_fun(year, pop, rule = 2)) %>%
+        group_by(year) %>%
+        mutate(value_share = pop / sum(pop, na.rm = T)) %>%
+        ungroup() %>%
+        select(province, year, value_share) %>%
+        left_join(L154.out_mpkm_R_trn_nonmotor_Yh, by = "year") %>%
```

Comment:
`rename` and `filter` on L154 before `left_join`
`left_join(L154.out_mpkm_R_trn_nonmotor_Yh %>% rename(value_mode = value) %>% filter(GCAM_region_ID == gcamchina.REGION_ID), by = "year")`